crates/db_schema/src/source/login_token.rs

```diff
@@ -0,0 +1,19 @@
+use crate::newtypes::LocalUserId;
+#[cfg(feature = "full")]
+use crate::schema::login_token;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "full", derive(Queryable, Identifiable))]
+#[cfg_attr(feature = "full", diesel(table_name = login_token))]
+pub struct LoginToken {
+  pub id: i32,
+  pub token: String,
+  pub user_id: LocalUserId,
+}
```

Comment:
usually i think you would want to add a few other fields here:

1. created timestamp 
2. ip address of the request that created the token
3. user agent of the request that created the token
4. invalidated timestamp (instead of deleting rows, use `invalidated is null` to filter validity)

this is all info you can (a) use to display a list of known sessions to the user and (b) so you can invalidate a session in some cases (e.g. geoip country of the session changes, user agent changes too much)

Reply:
@phiresky Good suggestions, I added them. Though Im doubtful about using invalidated timestamp, it means the table will grow forever and I dont see the benefit of keeping years old login info around. Adding another scheduled task for cleanup also doesnt seem worth the trouble.

Its true that storing IPs could cause some problems with privacy legislation. But I think its definitely worth it to fight abuse. We just need to be careful that IPs are not leaked to other users.

Reply:
instead of IP you could also store derived info ( city+country, ASIN) that's a bit less private and closer to what you'd want to use for session invalidation/display. But that would need an external service for the geolocation which is a pain. 

or you can store it truncated (e.g. the /24 for ipv4 and the /52 for ipv6)

Reply:
Yes geoip seems to complicated and error prone. Truncating would require the ip first. Im going to leave it like this and we can make changes in the future if necessary.